### PR TITLE
feat:update the lotus version

### PR DESCRIPTION
**Pull request by Terryhung**
## Changes
1. We need the [lotus](https://github.com/filecoin-project/lotus/pull/10446) change to prevent from causing excessive CPU usage.
2. Update the ffi version due to the lotus [changes](https://github.com/filecoin-project/lotus/commit/d2c1962e34e37cc49852d50ae2a8e4bd6ca66843).

### CODE REVIEW

**Comment by kasteph** [comment]
New release: https://github.com/filecoin-project/lotus/releases/tag/v1.20.3

File: go.sum [hunk 1]
```diff
@@ -356,8 +356,8 @@ github.com/filecoin-project/go-statestore v0.2.0 h1:cRRO0aPLrxKQCZ2UOQbzFGn4WDNd
 github.com/filecoin-project/go-statestore v0.2.0/go.mod h1:8sjBYbS35HwPzct7iT4lIXjLlYyPor80aU7t7a/Kspo=
 github.com/filecoin-project/go-storedcounter v0.1.0 h1:Mui6wSUBC+cQGHbDUBcO7rfh5zQkWJM/CpAZa/uOuus=
 github.com/filecoin-project/go-storedcounter v0.1.0/go.mod h1:4ceukaXi4vFURIoxYMfKzaRF5Xv/Pinh2oTnoxpv+z8=
-github.com/filecoin-project/lotus v1.20.1-0.20230302000218-dbbcf4b2ee96 h1:qtyxiKpJcsktp2EeL/OscVNbw1fRy8C5j9yeguA8AZY=
-github.com/filecoin-project/lotus v1.20.1-0.20230302000218-dbbcf4b2ee96/go.mod h1:IgwxSgrL1HgkSexOjTHOIdaTUIvPOetuAjAqFQgHkSc=
+github.com/filecoin-project/lotus v1.20.1-0.20230310193415-9d994dac9e65 h1:OfMFW/rkaLcWCwkKoveLwmwLgcoFg+IfVRtmoi64ofo=
```
Comment (kasteph) [inline go.sum:359]: New release here :) https://github.com/filecoin-project/lotus/releases/tag/v1.20.3

File: commands/daemon.go [hunk 2]
```diff
@@ -255,16 +254,13 @@ Note that jobs are not persisted between restarts of the daemon. See
 			node.Override(new(*schedule.Scheduler), schedule.NewSchedulerDaemon),
 			node.Override(new(*storage.Catalog), modules.NewStorageCatalog),
 			node.Override(new(*distributed.Catalog), modules.NewQueueCatalog),
-			node.Override(new(*lutil.CacheConfig), modules.CacheConfig(cacheFlags.BlockstoreCacheSize, cacheFlags.StatestoreCacheSize)),
```
Comment (frrist) [inline commands/daemon.go:258]: Don't remove these. They are required.

File: commands/daemon.go [hunk 3]
```diff
@@ -255,16 +254,13 @@ Note that jobs are not persisted between restarts of the daemon. See
...
 			// End Injection
 
 			node.Override(new(dtypes.Bootstrapper), isBootstrapper),
 			node.Override(new(dtypes.ShutdownChan), shutdown),
 			node.Base(),
 			node.Repo(r),
 
-			node.Override(new(dtypes.UniversalBlockstore), modules.NewCachingUniversalBlockstore),
```
Comment (frrist) [inline commands/daemon.go:266]: Don't remove this.

File: lens/lily/impl.go [hunk 4]
```diff
@@ -60,7 +60,6 @@ type LilyNodeAPI struct {
 	Scheduler *schedule.Scheduler
 
 	ExecMonitor stmgr.ExecMonitor
-	CacheConfig *util.CacheConfig
```
Comment (frrist) [inline lens/lily/impl.go:63]: Revert this change.

File: tasks/survey/minerprotocols/minerprotocols.go [hunk 5]
```diff
@@ -171,6 +171,12 @@ func fetchMinerProtocolModel(ctx context.Context, api API, addr address.Address,
 		return
 	}
 
+	// convert the protos to string
+	protosStrArray := []string{}
+	for _, v := range protos {
```
Comment (frrist) [inline tasks/survey/minerprotocols/minerprotocols.go:176]: Is this necessary?

File: commands/daemon.go (hunk 2 above)
Comment (Terryhung) [reply commands/daemon.go:258]: But there is no `CachingBlockstore` in lotus anymore. How to fix it?

File: lens/lily/impl.go (hunk 4 above)
Comment (Terryhung) [reply lens/lily/impl.go]: Sorry let me revert it back

File: go.mod [hunk 6]
```diff
@@ -11,10 +11,10 @@ require (
 	github.com/filecoin-project/go-amt-ipld/v3 v3.1.1 // indirect
```
Comment (frrist) [inline go.mod:11]: Update the version of go used by the project, its line 3 of this file

File: lens/lily/modules/statemanager.go [hunk 7]
```diff
@@ -22,7 +22,7 @@ var log = logging.Logger("lily/lens")
 var ErrExecutionTraceNotFound = fmt.Errorf("failed to find execution trace")
 
 func StateManager(lmctx helpers.MetricsCtx, lc fx.Lifecycle, cs *store.ChainStore, exec stmgr.Executor, sys vm.SyscallBuilder, us stmgr.UpgradeSchedule, bs beacon.Schedule, em stmgr.ExecMonitor) (*stmgr.StateManager, error) {
-	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em)
+	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em, nil)
```
Comment (frrist) [inline lens/lily/modules/statemanager.go:25]: what is the `nil` value being passed to the `stmgr`? I assume it's a datastore? The statemanager will likely need a concrete value passed here to operate.

File: .golangci.yml [hunk 8]
```diff
@@ -40,6 +40,9 @@ issues:
     - path: schemas/.*
       linters:
         - misspell
+    - linters:
+        - staticcheck
+      text: "SA1019:"
```
Comment (frrist) [inline .golangci.yml:45]: what does this change do?

File: lens/util/store.go [hunk 9]
```diff
@@ -70,6 +70,10 @@ func (cs *CachingBlockstore) HashOnRead(enabled bool) {
 	cs.blocks.HashOnRead(enabled)
 }
 
+func (cs *CachingBlockstore) Flush(ctx context.Context) error {
```
Comment (frrist) [inline lens/util/store.go:73]: Does this satisfy a new interface requirement?

**Review by frrist — CHANGES_REQUESTED** [review]
Left some more questions. https://github.com/filecoin-project/lily/pull/1153/files#diff-35818035b39b465ad281977acde860ea419bf987ddca1203c34f6904e3227dd7R25 will need to be addressed before this is ready for merge.

File: lens/util/store.go (hunk 9 above)
Comment (Terryhung) [reply lens/util/store.go:73]: Here is the Flush in datasotre: https://github.com/filecoin-project/lotus/blob/master/blockstore/blockstore.go#L114.

And [here](https://github.com/filecoin-project/lotus/blob/master/blockstore/blockstore.go#L17-L31) is the interface. Is it correct? thanks

File: .golangci.yml (hunk 8 above)
Comment (Terryhung) [reply .golangci.yml:45]: we skip the lint check for old package: `commands/daemon.go:6:2: SA1019: package io/ioutil is deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)`

[old lint logs](https://app.circleci.com/pipelines/github/filecoin-project/lily/4109/workflows/455c81c2-a098-4524-9676-abef485dd8d3/jobs/20185).

File: lens/lily/modules/statemanager.go (hunk 7 above)
Comment (Terryhung) [reply lens/lily/modules/statemanager.go:25]: Got it. But I am not familiar with it here. Could you help me to do it here?  thanks!

File: .golangci.yml (hunk 8 above)
Comment (frrist) [reply .golangci.yml:45]: Instead of silencing the linter errors, replace the deprecated dependency: `io/ioutil` with either the `io` or `os` package.

File: lens/util/store.go (hunk 9 above)
Comment (frrist) [reply lens/util/store.go:73]: Yup looks good, thanks.

File: lens/lily/modules/statemanager.go (hunk 7 above)
Comment (frrist) [reply lens/lily/modules/statemanager.go:25]: Here is the constructor from lotus: https://github.com/filecoin-project/lotus/blob/master/node/modules/stmgr.go#L13

Comment (frrist) [reply lens/lily/modules/statemanager.go:25]: So this should look something like:

```go
func StateManager(lmctx helpers.MetricsCtx, lc fx.Lifecycle, cs *store.ChainStore, exec stmgr.Executor, sys vm.SyscallBuilder, us stmgr.UpgradeSchedule, bs beacon.Schedule, em stmgr.ExecMonitor, metadataDs dtypes.MetadataDS) (*stmgr.StateManager, error) {
	sm, err := stmgr.NewStateManagerWithUpgradeScheduleAndMonitor(cs, exec, sys, us, bs, em, metadataDs)
```

Comment (frrist) [reply lens/lily/modules/statemanager.go:25]: This is the package lotus and lily use for dep injection like this: https://github.com/uber-go/fx

File: .golangci.yml (hunk 8 above)
Comment (Terryhung) [reply .golangci.yml]: Fixed!

File: lens/lily/modules/statemanager.go (hunk 7 above)
Comment (Terryhung) [reply lens/lily/modules/statemanager.go]: Fixed! Thanks for your help!

File: go.mod (hunk 6 above)
Comment (Terryhung) [reply go.mod:11]: Fixed!

**Review by frrist — APPROVED** [review]
